drivers/char/rust_example.rs

```diff
@@ -51,6 +53,106 @@ module! {
     },
 }
 
+const MAX_TOKENS: usize = 3;
+
+struct SharedStateInner {
+    token_count: usize,
+}
+
+struct SharedState {
+    state_changed: CondVar,
+    inner: Mutex<SharedStateInner>,
+}
+
+impl SharedState {
+    fn try_new() -> KernelResult<Arc<Self>> {
+        let state = Arc::try_new(Self {
+            // SAFETY: Init is called below.
```

Comment:
// SAFETY: `condvar_init!` is called below.

Reply:
Done.